reference/useful_queries.rst

```diff
@@ -0,0 +1,109 @@
+
+Detecting locks
+---------------
+
+This query will run across all worker nodes and identify locks, how long they've been open, and the offending query:
+
+.. code-block:: sql
+
+    SELECT run_command_on_workers($cmd$SELECT array_agg(blocked_statement || ' $ ' || cur_stmt_blocking_proc || ' $ ' ||cnt::text || ' $ ' || age) FROM (
+    SELECT   blocked_activity.query    AS blocked_statement,
+             blocking_activity.query   AS cur_stmt_blocking_proc,
+             count(*) cnt,
+             --now() - min(current_timestamp) AS how_long,
+             age(now(), min(blocked_activity.query_start)) AS "age"
+       FROM  pg_catalog.pg_locks         blocked_locks
+        JOIN pg_catalog.pg_stat_activity blocked_activity  ON blocked_activity.pid = blocked_locks.pid
+        JOIN pg_catalog.pg_locks         blocking_locks
+            ON blocking_locks.locktype = blocked_locks.locktype
+            AND blocking_locks.DATABASE IS NOT DISTINCT FROM blocked_locks.DATABASE
+            AND blocking_locks.relation IS NOT DISTINCT FROM blocked_locks.relation
+            AND blocking_locks.page IS NOT DISTINCT FROM blocked_locks.page
+            AND blocking_locks.tuple IS NOT DISTINCT FROM blocked_locks.tuple
+            AND blocking_locks.virtualxid IS NOT DISTINCT FROM blocked_locks.virtualxid
+            AND blocking_locks.transactionid IS NOT DISTINCT FROM blocked_locks.transactionid
+            AND blocking_locks.classid IS NOT DISTINCT FROM blocked_locks.classid
+            AND blocking_locks.objid IS NOT DISTINCT FROM blocked_locks.objid
+            AND blocking_locks.objsubid IS NOT DISTINCT FROM blocked_locks.objsubid
+            AND blocking_locks.pid != blocked_locks.pid
+        JOIN pg_catalog.pg_stat_activity blocking_activity ON blocking_activity.pid = blocking_locks.pid
+       WHERE NOT blocked_locks.GRANTED
+         AND blocking_locks.GRANTED
+       GROUP BY blocked_activity.query,
+                blocking_activity.query
+       ORDER BY 4
+    )a$cmd$);
+
+
+Querying size of your shards
+----------------------------
+
+This query will provide you with the size of every shard of a given distributed table, designated here with the placeholder :code:`my_distributed_table`:
+
+.. code-block:: sql
+
+    SELECT pg_size_pretty(result::bigint) 
+    FROM run_command_on_shards('my_distributed_table', $cmd$
+      SELECT pg_table_size('%s');
+    $cmd$);
+
+For more info about distributed table size, see :ref:`table_size`.
+
+Identifying unused indexes
+--------------------------
+
+This query will run across all worker nodes and identify any unused indexes for a given distributed table, designated here with the placeholder :code:`my_distributed_table`:
+
+.. code-block:: sql
+
+  SELECT *
+  FROM run_command_on_shards('my_distributed_table', $cmd$
```

Comment:
I adjusted this query so that it now runs. However I can't tell whether it works correctly. It indicated that my distributed table had no unused indices. I then created an index that seemed unused and re-ran the query but no change. Perhaps `pg_stat_user_indexes` had not caught up yet...

Reply:
@begriffs What changes did you make to the query? Could you try rerunning the query to see if the results are now different?